optimade/models/structures.py

```diff
@@ -646,15 +646,24 @@ def required_if_dimension_types_has_one(cls, v, values):
                 f"lattice_vectors is REQUIRED, since dimension_types is not [0, 0, 0] but is {values.get('dimension_types', 'Not specified')}"
             )
 
-        for dim_type, vector in zip(values.get("dimension_types", []), v):
+        for dim_type, vector in zip(values.get("dimension_types", (None,) * 3), v):
             if None in vector and dim_type == 1:
                 raise ValueError(
                     "Null entries in lattice vectors are only permitted when the corresponding dimension type is 0. "
-                    f"Here: dimension_types = {values.get('dimension_types', 'Not specified')}, lattice_vectors = {v}"
+                    f"Here: dimension_types = {tuple(getattr(_, 'value', None) for _ in values.get('dimension_types', []))}, lattice_vectors = {v}"
                 )
 
         return v
 
+    @validator("lattice_vectors")
+    def null_values_for_whole_vector(cls, v):
+        for vector in v:
+            if None in vector and any((isinstance(_, float) for _ in vector)):
+                raise ValueError(
+                    f"A lattice vector MUST be either all `null` or all numbers (vector: {vector}, all vectors: {v})"
```

Comment:
Is this true? The spec says:

> All three elements of the inner lists of floats MAY be null for non-periodic dimensions, i.e., those dimensions for which dimension_types is 0.

I agree that it makes sense for all null or no null to be the case...

Reply:
The spec is a little ambiguous here, for sure. My question is now if there are valid cases where a vector wouldn't be all numbers or all `null`?

Reply:
I think we take this out. I'd rather be less restrive then more when the spec is ambiguous.

Reply:
Sure - but let's leave this PR be for now then and add an issue for clarification in the spec.